lib/adjustable_income_calculator.rb

```diff
@@ -0,0 +1,99 @@
+class AdjustableIncomeCalculator
+  # https://www.ons.gov.uk/peoplepopulationandcommunity/birthsdeathsandmarriages/lifeexpectancies/bulletins/lifeexpectancyatbirthandatage65bylocalareasinenglandandwales/2015-11-04#national-life-expectancy-at-age-65
+  LIFE_EXPECTANCY = 84
+  TAX_FREE_POT_PORTION = 0.25
+  GROWTH_INTEREST_RATE = 0.03
+
+  def initialize(pot:, age:, desired_income:)
```

Comment:
Depending on the size of these and that they're unbounded, it's quite trivial to pretty much DoS the server.

We can take a closer look at this tomorrow. It might be a simple case of just validating that the pot size in particular does not exceed a super-huge amount.

I do wonder whether there may be a domain-specific reason why this wouldn't generally exceed say £10,000,000 or indeed any other arbitrarily large amount. Perhaps in this case the end-user is likely to have specialised accountancy  and wouldn't likely use this calculator?